build/llvm.ps1

```diff
@@ -0,0 +1,128 @@
+        mkdir $Env:CCACHE_DIR | Out-Null
+    }
+    $Env:CCACHE_DIR = Resolve-Path $Env:CCACHE_DIR
+    ccache -M 2G
+
+    Write-Vso "Found ccache config:"
+    ccache --show-config
+
+    $cacheRoot = $Env:CCACHE_DIR
+
+    Assert (Test-Path $cacheRoot) "$($cacheRoot) is missing"
+    Write-Vso "Using CCACHE_DIR: $($Env:CCACHE_DIR)"
+}
+else {
+    Write-Vso "Did not find ccache command"
+}
+
+Assert (Test-Path $llvmDir) "llvm-project submodule is missing"
+Assert (![string]::IsNullOrWhiteSpace($Env:PKG_NAME)) "PKG_NAME is not set"
+
+Assert (Test-CommandExists "cmake") "CMAKE not found"
+Assert (Test-CommandExists "ninja") "Ninja-Build not found"
+if($IsLinux) {
+    Assert (Test-CommandExists "docker") "Docker not found"
+}
+
+if(!(Test-Path $llvmBuildDir)) {
+    mkdir $llvmBuildDir | Out-Null
+}
+
+Write-Vso "Generating package: $($Env:PKG_NAME)"
+
+if ($IsLinux) {
+    function Build-ContainerImage {
+
+        Write-Vso "Building container image manylinux-llvm-builder"
+        exec -wd $buildDir {
+            Get-Content manylinux.Dockerfile | docker build -t manylinux-llvm-builder --build-arg LLVM_BUILD_DIR="$llvmBuildDir" -
+        }
+    }
+
+    function Invoke-ContainerImage {
+        # Verify input files/folders and mounts exist
+        Assert (Test-Path $llvmCmakeFile) "$($llvmCmakeFile) is missing"
+        Assert (Test-Path $llvmDir) "$($llvmDir) is missing"
+        Assert (Test-Path $llvmBuildDir) "$($llvmBuildDir) is missing"
+
+        Assert (Test-Path $srcRoot) "$($srcRoot) is missing"
+
+        Write-Vso "Running container image:"
+        $command = "docker run --rm -t --user vsts -e PKG_NAME=$($Env:PKG_NAME) -e SOURCE_DIR=$srcRoot -e LLVM_CMAKEFILE=$llvmCmakeFile -e LLVM_DIR=$llvmDir -e LLVM_BUILD_DIR=$llvmBuildDir -e CCACHE_DIR=$cacheRoot -e CCACHE_CONFIGPATH=$cacheRoot -v $($srcRoot):$($srcRoot) -v $($cacheRoot):$($cacheRoot) manylinux-llvm-builder"
+        if(Test-Path env:\CMAKE_INSTALL_PREFIX) {
+            $command = "docker run --rm -t --user vsts -e CMAKE_INSTALL_PREFIX=$($Env:CMAKE_INSTALL_PREFIX) -e PKG_NAME=$($Env:PKG_NAME) -e SOURCE_DIR=$srcRoot -e LLVM_CMAKEFILE=$llvmCmakeFile -e LLVM_DIR=$llvmDir -e LLVM_BUILD_DIR=$llvmBuildDir -e CCACHE_DIR=$cacheRoot -e CCACHE_CONFIGPATH=$cacheRoot -v $($srcRoot):$($srcRoot) -v $($cacheRoot):$($cacheRoot) manylinux-llvm-builder"
+        }
+
+        Write-Vso $command
+        exec {
+            Invoke-Expression $command
+        }
+    }
+
+    Build-ContainerImage
+    Invoke-ContainerImage
+}
+else {
+
+    if ($IsWindows) {
+        # find VS root
+        $vswhere = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe"
+        $visualStudioPath = & $vswhere -prerelease -latest -property installationPath
+        Write-Output "vs located at: $visualStudioPath"
+
+        # Call vcvars64.bat and write the set calls to file
+        cmd.exe /c "call `"$visualStudioPath\VC\Auxiliary\Build\vcvars64.bat`" && set > %temp%\vcvars.txt"
```

Comment:
Is the `vcvars.txt` file used again, or can this be piped straight into `ForEach-Object`?